lib/api/1.1/northbound/templates.js

```diff
@@ -32,8 +34,12 @@ function templatesRouterFactory (
      */
 
     router.get('/templates/library', function (req, res) {
-        presenter(req, res)
-            .render(templates.getAll());
+        Promise.map(templates.getAll(), function(t) {
+            return templates.get(t.name);
+        })
+        .then(function(t) {
+            presenter(req, res).render(t);
```

Comment:
@benbp Presenter.render can handle arrays.  I think the method name `render` may be a little misleading because it doesn't actually do any rendering; it ultimately just deals with promises and calls res.json.